### PR TITLE
Option to modify stiffness for set insertion point

### LINKED ISSUE #245: ETABS_Toolkit: Option to modify stiffness for Insertion point
#### Description:

ETABS allowes one to choose whether to change the stiffness modifier when changing the Insertion Point for a bar, I would like to be able to do this from BHoM.

I fact I already have a working version of ETABS with this functionality, (and working very similarly to the current Insertion Point implementation). So with @IsakNaslundBh blessing I'll make the PR

### PR DESCRIPTION
**Pull request by kThorsager**
### Issues addressed by this PR

 Closes #245 
allows one to access this: 
image

in BHoM, 

 ### Test files

https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/ETABS_Toolkit/Issue_245%20ModifyStiffness%20for%20SetInsertionPoint/Issue_245%20ModifyStiffness%20for%20SetInsertionPoint.gh?csf=1&e=Bckfd8

 ### Additional comments

### CODE REVIEW

**Comment by rwemay** [comment]
Great use of CustomData and the adapters to quickly facilitate more functionality. Purely as an aside, it would be good to continue the conversation on exposing more of this in the UI so that a user can find/see what is available to set, even if software/adapter specific. I've used enums in the Robot_Toolkit to help facilitate this, as well as creating more formal 'objects'. There's obviously no single solution, but good to start thinking about something global as we put more functionality into adapters/toolkits. @IsakNaslundBh @al-fisher @alelom

**Comment by IsakNaslundBh** [comment]
> Great use of CustomData and the adapters to quickly facilitate more functionality. Purely as an aside, it would be good to continue the conversation on exposing more of this in the UI so that a user can find/see what is available to set, even if software/adapter specific. I've used enums in the Robot_Toolkit to help facilitate this, as well as creating more formal 'objects'. There's obviously no single solution, but good to start thinking about something global as we put more functionality into adapters/toolkits. @IsakNaslundBh @al-fisher @alelom

Agree with you points here @rwemay .

For this @kThorsager added a boolean to go together with an enum following how it had been done before (the enum was there before the fragments was in place).
We had a quick chat though, and will switch this to adding a small fragement object to take care of the enum and boolean together (now that there are two properties to handle together) and add it as a fragment instead of custom data. Kalle will also fix the auto length offsets in the PR to be working with fragments instead of custom data. 

Also something good to go through in general in this milestone, the switch from custom data to fragments!

File: ETABS_Engine/Modify/SetInsertionPoint.cs [hunk 1]
```diff
@@ -39,11 +41,14 @@ public static partial class Modify
 
         public static Bar SetInsertionPoint(this Bar bar, BarInsertionPoint barInsertionPoint = BarInsertionPoint.Centroid)
         {
-            Bar clone = (Bar)bar.GetShallowClone();
+            return bar.SetInsertionPoint(barInsertionPoint);
```
Comment (IsakNaslundBh) [inline ETABS_Engine/Modify/SetInsertionPoint.cs]: Will this not lead to stack overflow?

Comment (kThorsager) [reply ETABS_Engine/Modify/SetInsertionPoint.cs]: Yes it will

**Pull request merged**